src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -64,17 +64,26 @@ public void removeRunningTask(Protos.TaskID taskId) {
   public void updateTaskForStatus(Protos.TaskStatus status) {
     //Case of name node, update the task map
     if (status.getTaskId().getValue().contains(HDFSConstants.NAME_NODE_TASKID)) {
+      //If initializing the first NN or reconciling the first NN or bootstrapping the first NN
+      //set the status to initialized
       if (status.getMessage().equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)
           || (currentAcquisitionPhase.equals(AcquisitionPhase.RECONCILING_TASKS)
-          && !isNameNode1Initialized())) {
+              && !isNameNode1Initialized())
```

Comment:
How do you know to set initialized=true for a reconciliation status update? Seems like you could be reconciling before NN1 was actually initialized.